### ISSUE TITLE
How to get signup form field values in signUp event?

### ISSUE THREAD

**Issue by tomaszs** [aws-amplify/amplify-js#3992]
I am using amplify-authenticator to handle user signup.

During signup user provides his name, email etc.

I need to call 3rd party endpoint with these data after signup.

The code looks like this:

```
const listener = (data: any) => {
        switch (data.payload.event) {
            case 'signUp':
                this.call3rdpartyAPI();
                break;
        }
    };

    Hub.listen('auth', listener);
```

How to get name and email typed into sign up form by the user?

**Comment by elorzafe** [comment]
Hi @tomaszs, on `Hub.listen` for signUp event you can only read username attribute. If you want to see all attributes after signup you can do add a post confirmation trigger on the userpool. You can add it by using amplify-cli, more info [here](https://aws-amplify.github.io/docs/cli-toolchain/quickstart#adding-a-lambda-trigger)

**Comment by tomaszs** [comment]
@elorzafe Yes, i have tried Hub.listen and can confirm there is only username. How to add post confirmation trigger after i have Amplify set up and running?

**Comment by elorzafe** [comment]
If you use [amplify-cli](https://aws-amplify.github.io/docs/cli-toolchain/quickstart#adding-a-lambda-trigger), when you add the **Post confirmation** trigger there is an option to edit there the function. Maybe [this forum thread](https://forums.developer.amazon.com/questions/95692/invoking-a-rest-api-from-within-a-lambda-function.html) can give you some guidance

**Comment by tomaszs** [comment]
So as i understand the only supported solution now is that i should trigger a Lambda and than from Lambda call whatever i like?

Is there any concept behind isolating UI from login process? Or is this just a missing feture?

**Comment by elorzafe** [comment]
> So as i understand the only supported solution now is that i should trigger a Lambda and than from Lambda call whatever i like?

Currently yes, a Lambda trigger is the way to do it.

> Is there any concept behind isolating UI from login process? Or is this just a missing feature?

I think this is a valid feature-request, I will label it as need discussion as well.

Thanks for your feedback! @tomaszs

**Comment by tomaszs** [comment]
Thanks!
